### PR TITLE
WIP: add PeerID lookup

### PR DESCRIPTION
**Pull request by keremgocen**
- [x] enhance key lookup logic and enable PeerID lookup in addition to the key name
- [x] write sharness tests

as discussed in #3807

License: MIT
Signed-off-by: Kerem Gocen <keremgocen@gmail.com>

### CODE REVIEW

File: core/commands/publish.go [hunk 1]
```diff
@@ -176,3 +181,36 @@ func publish(ctx context.Context, n *core.IpfsNode, k crypto.PrivKey, ref path.P
 		Value: ref.String(),
 	}, nil
 }
+
+func keylookup(k string, n *core.IpfsNode) (crypto.PrivKey, error) {
+
+	res, err := n.GetKey(k)
+	if res != nil {
```
Comment (Kubuxu) [inline core/commands/publish.go]: I would check here if error is different than `ErrNoSuchKey` if that is a case It signals different error than should happen here.

File: core/commands/publish.go [hunk 2]
```diff
@@ -176,3 +181,36 @@ func publish(ctx context.Context, n *core.IpfsNode, k crypto.PrivKey, ref path.P
 		Value: ref.String(),
 	}, nil
 }
+
+func keylookup(k string, n *core.IpfsNode) (crypto.PrivKey, error) {
```
Comment (Kubuxu) [inline core/commands/publish.go]: Please make the Node first argument.

**Comment by keremgocen** [comment]
Force pushed on my fork to change commit message for f93e8c58d8b92e1e8d998fb9525e99d187a6dced, should I close this for another PR @Kubuxu ?

**Comment by Kubuxu** [comment]
No, force pushes are OK.

**Comment by keremgocen** [comment]
Do you have any recommendations for how to fix the failing CI [here](https://travis-ci.org/ipfs/go-ipfs/builds/228194943?utm_source=github_status&utm_medium=notification)? Also a review is required to proceed this one forward I see.

**Comment by Kubuxu** [comment]
@keremgocen don't worry about travis too much (it likes to fail on its own, we load their machines a bit too much).

**Comment by whyrusleeping** [comment]
@Kubuxu :+1:  here?

**Pull request merged**